torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -251,7 +251,18 @@ Tensor detach(const Tensor & self) {
 Tensor & detach_(Tensor & self) {
   RECORD_FUNCTION("detach_", std::vector<c10::IValue>({self}));
   if (self.is_view()) {
-    AT_ERROR("Can't detach views in-place. Use detach() instead");
+    // NB: is_view() ==> get_autograd_meta()
+    auto diff_view_meta = static_cast<torch::autograd::DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(self));
+    // See NOTE [ View + Inplace detection ]
+    if (diff_view_meta->creation_meta == CreationMeta::MULTI_OUTPUT_SAFE) {
+        TORCH_WARN("This view is an output of a function that "
+                   "returns multiple views. Detaching such views inplace "
+                   "is being deprecated and will be forbidden "
+                   "starting from version 1.8. Consider using `unsafe_` "
```

Comment:
I think you want to recommend the out of place version here, there is no unsafe

Reply:
Done